src/main/java/org/jabref/JabRefMain.java

```diff
@@ -93,58 +98,43 @@ private static void ensureCorrectJavaVersion() {
-        PreferencesMigrations.upgradePrefsToOrgJabRef();
-        PreferencesMigrations.upgradeSortOrder();
-        PreferencesMigrations.upgradeFaultyEncodingStrings();
-        PreferencesMigrations.upgradeLabelPatternToBibtexKeyPattern();
-        PreferencesMigrations.upgradeImportFileAndDirePatterns();
-        PreferencesMigrations.upgradeStoredCustomEntryTypes();
-        PreferencesMigrations.upgradeKeyBindingsToJavaFX();
-        PreferencesMigrations.addCrossRefRelatedFieldsForAutoComplete();
-        PreferencesMigrations.upgradeObsoleteLookAndFeels();
+        // Perform migrations
+        migratePreferences();
 
-        FallbackExceptionHandler.installExceptionHandler();
-
-        ensureCorrectJavaVersion();
-
-        ProxyPreferences proxyPreferences = preferences.getProxyPreferences();
-        ProxyRegisterer.register(proxyPreferences);
-        if (proxyPreferences.isUseProxy() && proxyPreferences.isUseAuthentication()) {
-            Authenticator.setDefault(new ProxyAuthenticator());
-        }
+        configureProxy(preferences.getProxyPreferences());
 
         Globals.startBackgroundTasks();
 
-        // Update handling of special fields based on preferences
-        InternalBibtexFields
-                .updateSpecialFields(Globals.prefs.getBoolean(JabRefPreferences.SERIALIZESPECIALFIELDS));
-        // Update name of the time stamp field based on preferences
-        InternalBibtexFields.updateTimeStampField(Globals.prefs.getTimestampPreferences().getTimestampField());
-        // Update which fields should be treated as numeric, based on preferences:
-        InternalBibtexFields.setNumericFields(Globals.prefs.getStringList(JabRefPreferences.NUMERIC_FIELDS));
-
-        // Read list(s) of journal names and abbreviations
-        Globals.journalAbbreviationLoader = new JournalAbbreviationLoader();
-
-        /* Build list of Import and Export formats */
-        Globals.IMPORT_FORMAT_READER.resetImportFormats(Globals.prefs.getImportFormatPreferences(),
-                Globals.prefs.getXMPPreferences(), Globals.getFileUpdateMonitor());
-        EntryTypes.loadCustomEntryTypes(preferences.loadCustomEntryTypes(BibDatabaseMode.BIBTEX),
-                preferences.loadCustomEntryTypes(BibDatabaseMode.BIBLATEX));
-        Globals.exportFactory = Globals.prefs.getExporterFactory(Globals.journalAbbreviationLoader);
+        applyPreferences(preferences);
 
         // Process arguments
         ArgumentProcessor argumentProcessor = new ArgumentProcessor(args, ArgumentProcessor.Mode.INITIAL_START);
 
-        // Initialize protected terms loader
-        Globals.protectedTermsLoader = new ProtectedTermsLoader(Globals.prefs.getProtectedTermsPreferences());
-
         // Check for running JabRef
+        if (!allowMultipleAppInstances(args)) {
+            return;
+        }
+
+        // See if we should shut down now
+        if (argumentProcessor.shouldShutDown()) {
+            Globals.shutdownThreadPools();
```

Comment:
This code is appearing twice, too. Can it be moved to `shutdownCurrentInstance()`?

Alternatively, change line this line to 

   if (!allowMultipleAppInstances(args) || argumentProcessor.shouldShutDown()) {

and remove the shutdown code from `allowMultipleAppInstances`.